tls/extensions/s2n_client_max_frag_len.c

```diff
@@ -53,11 +53,25 @@ static int s2n_client_max_frag_len_recv(struct s2n_connection *conn, struct s2n_
 
     uint8_t mfl_code;
     POSIX_GUARD(s2n_stuffer_read_uint8(extension, &mfl_code));
-    if (mfl_code > S2N_TLS_MAX_FRAG_LEN_4096 || mfl_code_to_length[mfl_code] > S2N_TLS_MAXIMUM_FRAGMENT_LENGTH) {
+
+    /*
+     *= https://tools.ietf.org/rfc/rfc6066#section-4
+     *# If a server receives a maximum fragment length negotiation request
+     *# for a value other than the allowed values, it MUST abort the
+     *# handshake with an "illegal_parameter" alert.
```

Comment:
I don't see how this aborts the handshake with an alert? Maybe you should add a comment to this RFC comment as it isn't clear what's happening here.

Reply:
Hmmm that's not specific to this code/comment, and I'm not sure if we should discuss it every time the RFC mentions alerts. That might get a bit out of hand.

Basically, we never follow the specific alert recommendations in the RFCs and prefer to just close/notify. The actual "abort the handshake" is handled in s2n_handshake_io via WITH_ERROR_BLINDING/s2n_connection_kill. If you're inside a read handler in S2N, "abort the handshake with an error" == "return from the method with an error".

Reply:
This is just returning S2N_SUCCESS though, so are we actually just ignoring max fragment requests larger than the allowed values?

Reply:
Oh my bad, I thought this was the server extension!

I've added an exception compliance comment. This is pretty common behavior in our extensions-- the S2N server seems to like to err on the side of ignoring extensions it doesn't like.